### PR TITLE
!!! Added ComponentHistory app

### PR DESCRIPTION
**Pull request by Sbalbp**
App to show the history of installations of components made on different hosts.

### CODE REVIEW

**Comment by atsareg** [comment]
Zoltan, please confirm this PR

File: WebApp/handler/ComponentHistoryHandler.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+  AUTH_PROPS = 'authenticated'
+
+  @asyncGen
+  def web_getInstallationData( self ):
+
+    req = self.__request()
+
+    client = ComponentMonitoringClient()
+    result = client.getInstallations( req[ 'installation' ], req[ 'component' ], req[ 'host' ], True )
```
Comment (zmathe) [inline WebApp/handler/ComponentHistoryHandler.py]: This call will block Tornado.
result = yield self.threadTask(client.getInstallations, req['installation'], req['component'], req['host'], True)

File: WebApp/handler/ComponentHistoryHandler.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                         'System': installation[ 'Component' ][ 'System' ], \
+                         'Type': installation[ 'Component' ][ 'Type' ], \
+                         'Installed': installation[ 'InstallationTime' ].strftime( '%Y-%m-%d %H:%M' ), \
+                         'Uninstalled': uninstalled } )
+      timestamp = Time.dateTime().strftime( '%Y-%m-%d %H:%M [UTC]' )
+      total = len( installations )
+      callback = { 'success' : 'true', 'result' : values,
+                'total' : total, 'date' : timestamp }
+    else:
+      callback = { 'success' : 'false' , 'error' : result[ 'Message' ] }
```
Comment (zmathe) [inline WebApp/handler/ComponentHistoryHandler.py]: WErr.fromSERROR(result)
I know I also use this format, but I plan to replace to WErr

File: WebApp/handler/ComponentHistoryHandler.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    userData = self.getSessionData()
+
+    setup = userData[ 'setup' ].split( '-' )[-1]
+    systemList = []
+    system = None
+
+    fields = [ 'name', 'host', 'module', 'system', 'type' ]
+
+    client = ComponentMonitoringClient()
+    result = client.getInstallations( {}, {}, {}, True )
```
Comment (zmathe) [inline WebApp/handler/ComponentHistoryHandler.py]: yield self.theradTask ...

File: WebApp/handler/ComponentHistoryHandler.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+      else:
+        self.pageNumber = 0
+    else:
+      self.numberOfInstallations = 25
+      self.pageNumber = 0
+
+    # Check every possible selector
+
+    if 'name' in self.request.arguments:
+      print self.request.arguments[ 'name' ][-1]
```
Comment (zmathe) [inline WebApp/handler/ComponentHistoryHandler.py]: delete the print

File: WebApp/handler/ComponentHistoryHandler.py [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+      req[ 'installation' ][ 'Instance' ] = list( json.loads( self.request.arguments[ 'name' ][-1] ) )
+
+    if 'host' in self.request.arguments:
+      print self.request.arguments[ 'host' ][-1]
```
Comment (zmathe) [inline WebApp/handler/ComponentHistoryHandler.py]: delete

File: WebApp/handler/ComponentHistoryHandler.py [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+      req[ 'host' ][ 'HostName' ] = list( json.loads( self.request.arguments[ 'host' ][-1] ) )
+
+    if 'system' in self.request.arguments:
+      print self.request.arguments[ 'system' ][-1]
```
Comment (zmathe) [inline WebApp/handler/ComponentHistoryHandler.py]: delete

File: WebApp/handler/ComponentHistoryHandler.py [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+      req[ 'component' ][ 'System' ] = list( json.loads( self.request.arguments[ 'system' ][-1] ) )
+
+    if 'module' in self.request.arguments:
+      print self.request.arguments[ 'module' ][-1]
```
Comment (zmathe) [inline WebApp/handler/ComponentHistoryHandler.py]: delete

File: WebApp/handler/ComponentHistoryHandler.py [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+      req[ 'component' ][ 'Module' ] = list( json.loads( self.request.arguments[ 'module' ][-1] ) )
+
+    if 'type' in self.request.arguments:
+      print self.request.arguments[ 'type' ][-1]
```
Comment (zmathe) [inline WebApp/handler/ComponentHistoryHandler.py]: delete

File: WebApp/static/DIRAC/ComponentHistory/classes/ComponentHistory.js [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+
+       if (GLOBAL.VIEW_ID == "desktop") {
+
+         me.launcher.title = "Component history";
+         me.launcher.maximized = false;
+
+         var oDimensions = GLOBAL.APP.MAIN_VIEW.getViewMainDimensions();
+
+         me.launcher.width = oDimensions[0];
+         me.launcher.height = oDimensions[1] - GLOBAL.APP.MAIN_VIEW.taskbar.getHeight();
```
Comment (zmathe) [inline WebApp/static/DIRAC/ComponentHistory/classes/ComponentHistory.js]: me.launcher.height = oDimensions[1];

File: WebApp/static/DIRAC/ComponentHistory/classes/ComponentHistory.js [hunk 10]
```diff
@@ -0,0 +1,185 @@
...
+         scope : me
+       });
+
+       me.leftPanel.setGrid(me.grid);
+
+       me.add([ me.leftPanel, me.grid ]);
+     },
+     __executeAction : function(action){
+       var me = this;
+       GLOBAL.APP.CF.alert(action+" button pressed","info");
```
Comment (zmathe) [inline WebApp/static/DIRAC/ComponentHistory/classes/ComponentHistory.js]: it is not implemented...

**Comment by fstagni** [comment]
I can't comment on the content of the code, so I just say that some comments, and a docstring for the handler part would be welcome.

**Comment by zmathe** [comment]
The code is okay.

**Pull request merged**